pkg/northbound/gnmi/set.go

```diff
@@ -313,14 +333,70 @@ func (s *Server) doDelete(prefix *gnmi.Path, u *gnmi.Path,
 	if prefixPath != "/" {
 		path = fmt.Sprintf("%s%s", prefixPath, path)
 	}
-	// Checks for read only paths
-	_, err := findPathFromModel(path, rwPaths, false)
-	if err != nil {
-		return nil, err
-	}
 	deletes = append(deletes, path)
 	return deletes, nil
+}
+
+// iterate through the updates and check that none of them include a `set` of a
+// readonly attribute - this is done by checking with the relevant model
+func (s *Server) checkForReadOnly(target string, deviceType devicetype.Type, version devicetype.Version,
```

Comment:
`checkForReadOnly` that was there previously has been replaced - I use the `findPathFromModel` instead, because it has the benefit of finding the `type` that the path refers to, and at the same time searches throught the set of Read-Write paths in the model - and so Read-Only paths will be rejected